### PR TITLE
Avalonia: Another Cleanup

### LINKED ISSUE #3491: Avalonia DLC Manager broken after 1.1.187 (Last working 1.1.185) NullReferenceException
### What's the issue you encountered?

NullReferenceException thrown when either of these conditions are met

* Opening the Manage DLC window for a game that already has DLC added
* Selecting 1 or more DLC in the file browser and adding them
* Clicking Remove, Remove All, or Save on the DLC manager screen for a title that has no DLC added.

These issues do not replicate under the original Ryujinx GUI, or on Avalonia 1.1.185

### How can the issue be reproduced?

1. Right click a title that already has DLC added
2. Select Manage DLC

or

1. Right click a title that has not yet had DLC added, 
2. Select Manage DLC
3. Click Add (or Remove, Remove All, Save)
4. Browse to and add a DLC file.

### Log file

[Ryujinx_1.1.187_2022-07-27_06-45-26.log](https://github.com/Ryujinx/Ryujinx/files/9193220/Ryujinx_1.1.187_2022-07-27_06-45-26.log)
[Ryujinx_1.1.193_2022-07-27_22-08-33.log](https://github.com/Ryujinx/Ryujinx/files/9198518/Ryujinx_1.1.193_2022-07-27_22-08-33.log)
[Ryujinx_1.1.193_2022-07-27_22-09-07.log](https://github.com/Ryujinx/Ryujinx/files/9198519/Ryujinx_1.1.193_2022-07-27_22-09-07.log)
[Ryujinx_1.1.193_2022-07-27_22-09-28.log](https://github.com/Ryujinx/Ryujinx/files/9198520/Ryujinx_1.1.193_2022-07-27_22-09-28.log)

### Environment?

- Ryujinx version: 1.1.187+</br>
- Game version: Irrelevant</br>
- System Specs:
  - OS: Windows 10
  - CPU: E5-1680 v2
  - GPU: GTX 1060
  - RAM: 32GB
- Applied Mods : None

### Additional context?

Additional info about your environment:</br>
Australian Locale with AU:English language pack
Issue replicates from clean install

DLC.json for one of several effected titles, Fire Emblem Three Houses - 010055d009f78000

> [
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC Side Story Cindered Shadows][010055D009F79001][v196608].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/8110c05d7151828750b8401605345ba8.nca",
>         "title_id": 72151946046705665,
>         "enabled": true
>       }
>     ]
>   },
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC Officers Academy Uniform Byleth][010055D009F79006][v262144].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/486f71d56be343fcc8be01ba0a227e44.nca",
>         "title_id": 72151946046705670,
>         "enabled": true
>       }
>     ]
>   },
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC New Attire][010055D009F79002][v262144].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/4f69c0282e2b3f5edbad3095a65b1240.nca",
>         "title_id": 72151946046705666,
>         "enabled": true
>       }
>     ]
>   },
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC Auxiliary Battle Maps][010055D009F79003][v196608].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/8bc72ba649bfd211eeacffd60f525340.nca",
>         "title_id": 72151946046705667,
>         "enabled": true
>       }
>     ]
>   },
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC Additional Support Items][010055D009F79005][v196608].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/6407e946907815288112412ba0ff09ca.nca",
>         "title_id": 72151946046705669,
>         "enabled": true
>       }
>     ]
>   },
>   {
>     "path": "W:\\Disk Images\\Switch\\Fire Emblem Three Houses\\DLC\\Fire Emblem Three Houses [DLC Additional Quests][010055D009F79004]v196608].nsp",
>     "dlc_nca_list": [
>       {
>         "path": "/3d390aebeb0745fc73dc6dc8fb00fc0c.nca",
>         "title_id": 72151946046705668,
>         "enabled": true
>       }
>     ]
>   }
> ]

### PR DESCRIPTION
**Pull request by AcK77**
This PR is a cleanup to the avalonia code recently added:

- Some XAML file are autoformatted like a previous PR.
- Dlc is renamed to DownloadableContent (Locale excluded).
- DownloadableContentManagerWindow is a bit improved (Fixes #3491).
- Remove duplicated code for AttachDevTools.
- Fixes a warning about Content control clashing with Avalonia.NameGenerator InitializeComponent.
- Some nits here and there.

### CODE REVIEW

**Comment by gdkchan** [comment]
Would be nice to fix the warning on `ContentDialogHelper.CreateWarningDialog(LocaleManager.Instance["UserProfileNoImageError"], "");` too while you're at it.

File: Ryujinx.Ava/Ui/Models/CheatModel.cs [hunk 1]
```diff
@@ -11,8 +11,8 @@ public class CheatModel : BaseModel
 
         public CheatModel(string name, string buildId, bool isEnabled)
         {
-            Name = name;
-            BuildId = buildId;
+            Name      = name;
```
Comment (gdkchan) [inline Ryujinx.Ava/Ui/Models/CheatModel.cs:14]: Note that those changes will be lost if the file is auto-formatted again.

**Review by gdkchan — COMMENTED** [review]
Build is failing.

**Comment by AcK77** [comment]
> Would be nice to fix the warning on `ContentDialogHelper.CreateWarningDialog(LocaleManager.Instance["UserProfileNoImageError"], "");` too while you're at it.

Already done in #3492

File: Ryujinx.Ava/Ui/Models/CheatModel.cs (hunk 1 above)
Comment (AcK77) [reply Ryujinx.Ava/Ui/Models/CheatModel.cs:14]: Yeah it's fine ;)

**Comment by Squall-Leonhart** [comment]
Check boxes default to a "-" when the DLC Manager is opened instead of a tick,

image

when you cycle the selection they become checks

image

**Comment by AcK77** [comment]
I haven’t touch anything related to the checkboxes, the issue is somewhere else ^^'

**Comment by Squall-Leonhart** [comment]
mm, ok, another post 1.1.187+ regression.

File: Ryujinx.Ava/Ui/Windows/DownloadableContentManagerWindow.axaml.cs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                    container = new DownloadableContentContainer
+                    {
+                        Path                       = downloadableContent.ContainerPath,
+                        DownloadableContentNcaList = new List<DownloadableContentNca>()
+                    };
+                }
+
+                container.DownloadableContentNcaList.Add(new DownloadableContentNca
+                {
+                    Enabled = downloadableContent.Enabled,
```
Comment (Squall-Leonhart) [inline Ryujinx.Ava/Ui/Windows/DownloadableContentManagerWindow.axaml.cs:245]: should this not be downloadableContent.IsEnabled,?

**Comment by Squall-Leonhart** [comment]
Its not actually the checkbox filling being wrong, the Window is not retrieving the actual state of the Enabled box for the listed DLC, so toggling the box doesn't update the DLC.json.

File: Ryujinx.Ava/Ui/Windows/DownloadableContentManagerWindow.axaml.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                    await ContentDialogHelper.CreateErrorDialog(LocaleManager.Instance["DialogDlcNoDlcErrorMessage"]);
+                }
+            }
+        }
+
+        private void RemoveDownloadableContents(bool removeSelectedOnly = false)
+        {
+            if (removeSelectedOnly)
+            {
+                DownloadableContents.RemoveAll(DownloadableContents.Where(x => x.Enabled).ToList());
```
Comment (Squall-Leonhart) [inline Ryujinx.Ava/Ui/Windows/DownloadableContentManagerWindow.axaml.cs:178]: think this should also be x.IsEnabled

**Comment by AcK77** [comment]
> Its not actually the checkbox filling being wrong, the Window is not retrieving the actual state of the Enabled box for the listed DLC, so toggling the box doesn't update the DLC.json.

Fixed

**Review by gdkchan — APPROVED** [review]
lgtm, thanks. But I don't see the point of aligning those assignments when it will probably be re-formatted again in the future.

**Review by Thog — APPROVED** [review]

**Pull request merged**